### ISSUE TITLE
delite/register: fails on iOS 8 when registering a custom element that inherits only from HTMLElement

### ISSUE THREAD

**Issue by AdrianVasiliu** [ibm-js/delite#259]
All unit tests of decor, delite and deliteful pass fine on iOS 8 beta 5 (tested on a physical iPhone 5S), with only a single exception: one of the test cases from the delite/register unit test. More exactly, this attempt to use the register() function:
```
TestWidget = register("test-simple-widget", [HTMLElement], { ... });
```
(see https://github.com/ibm-js/delite/blob/master/tests/unit/register.js#L55) fails on iOS 8 beta 5 with the following error:
```
[Error] FAIL: simple (1ms)
	/test/fail (console.js, line 40)
	dispatcher (aspect.js, line 91)
	emit (on.js, line 281)
	emit (Evented.js, line 31)
	publish (topic.js, line 25)
	handleTestError (Test.js, line 136)
...
[Error] Error: Attempted to assign to readonly property.
buildStubs@http://9.128.82.125:8020/dojo2-pure/dcl/mini.js:221:9
dcl@http://9.128.82.125:8020/dojo2-pure/dcl/mini.js:109:13
register@http://9.128.82.125:8020/dojo2-pure/delite/register.js:314:17
simple@http://9.128.82.125:8020/dojo2-pure/delite/tests/unit/register.js:39:25
...
```

The error message "Attempted to assign to readonly property" is of a similar kind as the dojo/on issue on iOS 8, fixed in https://github.com/dojo/dojo/commit/f164dc2c231aab806318055bbc3a6cd44674ee03. Now, while some other delite tests still use dojo/on directly, the test of delite/register does not. However, dojo/on is present in the picture because `intern` uses it. But this is not the 1.9.x version on which delite depends, it is 2.0.0-dev from https://github.com/csnover/dojo2-core, which differs in some extent. Although the error does not seem raised by this dojo/on code (it comes from dcl code), I tried to apply the same fix for the dojo/on issue as in dojo 1.9, but this didn't change anything.

Finally, a comparison with the other calls of register() in the same tests - calls which do work in iOS 8 - shows that what matters in the failing call is the fact that there is no other class than HTMLElement among the parent classes passed as argument. That is, changing the test to add Stateful as parent class:

```
TestWidget = register("test-simple-widget", [HTMLElement, Stateful], { ... });
```

makes it pass fine on iOS 8. 

So doing this change could be a way to fix the failure, but I'm not sure whether this wouldn't just "hide" another (deeper) issue. According to the current state of the tests and doc, delite/register.register() seems to impose only one constraint on the `superclasses` argument: the "first one must be [descendant] of HTMLElement" (quoting from the doc). The doc says that the method "declares a widget" while it does not say whether CustomElement, Stateful or Widget must be present among the superclasses. If it should work just with HTMLElement as parent class, we apparently need a fix in dcl. Otherwise, the doc may need to clarify the picture.

**Comment by wkeese** [comment]
Same failure on Mac Safari 8.

**Comment by wkeese** [comment]
Looks like this is fixed in iOS 9.3 (and presumably iOS 10), despite the fact that they never responded to your ticket.
